src/EndmemberLibrary.jl

```diff
@@ -238,62 +407,109 @@ function split_library(library::SpectralLibrary, split_fraction::Float64)
 
 end
 
+"""
+    reduce_endmembers_nmf!(library::SpectralLibrary, max_endmembers_per_class::Int64)
+
+Reduce the number of endmembers in the spectral library using Non-negative Matrix
+Factorization (NMF).
+
+- For each class in `library.class_valid_keys`, apply NMF to the spectra subset to identify
+the specified maximum number of endmembers.
+- Update `library.classes` and `library.spectra` to contain the reduced set of endmembers.
+
+# Notes
+- This function modifies the `library` in place.
+- Only `library.good_bands` will be considered in the NMF computation.
+- The NMF uses a maximum of 500 iterations with a tolerance of 1.0e-2 for convergence.
+"""
 function reduce_endmembers_nmf!(library::SpectralLibrary, max_endmembers_per_class::Int64)
 
     reduced_library = []
-    new_classes = copy(library.classes)[1:max_endmembers_per_class*size(library.class_valid_keys)[1]]
+    new_classes =
+        copy(library.classes)[1:max_endmembers_per_class*size(library.class_valid_keys)[1]]
     for (_c, cla) in enumerate(library.class_valid_keys)
-        library_subset = library.spectra[library.classes .== cla, library.good_bands]
+        library_subset = library.spectra[library.classes.==cla, library.good_bands]
 
         out_spectra = zeros(max_endmembers_per_class, size(library.spectra)[2])
-        out_spectra[:,:] .= NaN
+        out_spectra[:, :] .= NaN
 
         r = nnmf(library_subset, max_endmembers_per_class; maxiter=500, tol=1.0e-2)
-        out_spectra[:,library.good_bands] = r.H
+        out_spectra[:, library.good_bands] = r.H
 
-        push!(reduced_library,out_spectra)
-        new_classes[(_c-1) * max_endmembers_per_class + 1:_c * max_endmembers_per_class] .= cla
+        push!(reduced_library, out_spectra)
+        new_classes[(_c-1)*max_endmembers_per_class+1:_c*max_endmembers_per_class] .= cla
     end
-    library.spectra = cat(reduced_library...,dims=1)
+    library.spectra = cat(reduced_library..., dims=1)
     library.classes = new_classes
 end
 
+"""
+    reduce_endmembers_pca!(library::SpectralLibrary, max_endmembers_per_class::Int64)
+
+Reduce the number of endmembers in the spectral library using Principal Component Analysis
+(PCA).
+
+- For each class in `library.class_valid_keys`, apply PCA to the spectra subset to identify
+the specified maximum number of endmembers.
+- Update `library.classes` and `library.spectra` to contain the reduced set of endmembers.
+
+# Notes
+- This function modifies the `library` in place.
+- Only `library.good_bands` will be considered in the PCA computation.
```

Comment:
Should probably add a note about the parameter used in the PCA but I couldn't find a concise way of doing that... Something like:
` - The PCA is performed with maximum output dimension `max_endmembers_per_class` and with the ratio of variances preserved in the principal subspace set to 1.` ?

Reply:
How about:

```
The PCA is truncated to `max_endmembers_per_class` PCs per endmember class.
```

Reply:
Sounds good.